extension/js/agent/utils/serializeObject.js

```diff
@@ -90,13 +90,28 @@
       return this.classPropertyCache[info.name];
     }
 
-    var props = this.serializeObject(_.pick(serializeObject, info.keys));
+    // :FIXME: Properties should always be resolved on `object`.  This might
+    // break memoization though.
```

Comment:
actually, the reason why we resolve on the protochain is to do overrides well
think about a subclass or instance overriding a prop
@jiaweihli so in that case, i don't think we need the fixme